backend/cfg/cfg_available_regs.ml

```diff
@@ -0,0 +1,381 @@
+[@@@ocaml.warning "+a-30-40-41-42"]
+
+module DLL = Flambda_backend_utils.Doubly_linked_list
+module R = Reg
+module RAS = Reg_availability_set
+module RD = Reg_with_debug_info
+module V = Backend_var
+
+(* If permitted to do so by the command line flags, this pass will extend live
+   ranges for otherwise dead but available registers across allocations, polls
+   and calls when it is safe to do so. This allows the values of more variables
+   to be seen in the debugger, for example when the last use of some variable is
+   just before a call, and the debugger is standing in the callee. It may
+   however affect the semantics of e.g. finalizers. *)
+let extend_live () = !Dwarf_flags.gdwarf_may_alter_codegen
+
+(* CR xclerc for xclerc: consider passing this value through the domain. *)
```

Comment:
I agree in principle would rather keep the code
closer to the other pipeline; I have tweaked the
CR to mention "the context" rather than "the
domain".